checker/src/main/java/org/checkerframework/checker/nullness/CollectionToArrayHeuristics.java

```diff
@@ -140,30 +147,58 @@ private void setComponentNullness(boolean isNonNull, AnnotatedTypeMirror type) {
      * @param receiver the expression for the receiver collection
      * @return true if the argument is handled and assume to return nonnull elements
      */
-    private boolean isHandledArrayCreation(Tree argument, String receiver) {
-        if (!(argument instanceof NewArrayTree)) {
+    private boolean isHandledArrayCreation(ExpressionTree argument, String receiver) {
+        final NewArrayTree newArr;
+
```

Comment:
```suggestion
argument = TreeUtils.withoutParens(argument);
```

Maybe we should be more consistent about making sure parentheses don't mess up heuristics. Or is this already happening elsewhere?